snapshots/snapshot_stg_orders.sql

```diff
@@ -0,0 +1,34 @@
+{% snapshot snapshot_dim_customers %}
+{#
+environment aware target_schema
+  -- either use this config: 
+        target_schema=target.schema if target.name == 'dev' else 'prod_snapshot',
+  -- or
+        set a variable beforehand and call the variable
+    -- local in the sql file
+      -- ex: 
+        {% set target_snapshot_schema=target.schema if target.name == 'dev' else 'prod_snapshot' %}
+        config: target_schema=target_snapshot_schema
+    -- globally in dbt_project.yml
+      -- ex: 
+          vars:
+            target_snapshot_schema: "{{ target.schema if target.name == 'dev' else 'prod_snapshot' }}"
+          config: target_schema=var('target_snapshot_schema')
+
+#}
+
+{% set target_snapshot_schema=target.schema if target.name == 'dev' else 'prod_snapshot' %}
+
+{{
+    config(
+      target_schema=var('target_snapshot_schema'),
```

Comment:
I haven't done this before. Which value is used  - the value set in the dbt_project.yml or the value set on line 20? And again, same question about is this a recommended way to set up schemas for snapshots?

Reply:
In this case it was using the variable from the `dbt_project.yml`, but this is needlessly complex, so I am going to simplify.